go-controller/pkg/ovn/master.go

```diff
@@ -306,10 +307,10 @@ func (oc *DefaultNetworkController) syncNodes(kNodes []interface{}) error {
 		return fmt.Errorf("failed to get node logical switches which have other-config set: %v", err)
 	}
 
-	staleNodes := sets.NewString()
+	staleSwitches := sets.NewString()
 	for _, nodeSwitch := range nodeSwitches {
-		if nodeSwitch.Name != types.TransitSwitch && !foundNodes.Has(nodeSwitch.Name) {
-			staleNodes.Insert(nodeSwitch.Name)
+		if !strings.HasSuffix(nodeSwitch.Name, types.TransitSwitch) && !foundNodes.Has(nodeSwitch.Name) {
```

Comment:
nice catch, the perfect fix for long term is to use dbindexes and DNC shouldn't even be touching switches that don't belong to DNC.
I see an attempt to do something similar tot hat has been done in L302 where we ignore the switches that are not part of DNC there itself, is it that the L3 network transit switches are missing that externalID ? I think the predicate match should be changed instead of hacking the transit switch matches cause `nodeSwitches` should not contain the switches that don't belong to DNC. WDYT? cause it feels a bit icky to be considering a prefix we expect only for other network controllers here..
same for L2